keel-api/src/main/kotlin/com/netflix/spinnaker/keel/rest/ResourceController.kt

```diff
@@ -75,14 +61,13 @@ class ResourceController(
   }
 
   @PutMapping(
```

Comment:
Not sure what the semantically correct HTTP verb is for an upsert but I suspect `POST` might be more appropriate. `PUT` is specifically for updating an existing thing IIRC.

Reply:
I think it should be `PUT`: https://stackoverflow.com/questions/18470588/in-rest-is-post-or-put-best-suited-for-upsert-operation

Reply:
I read the same thing, but specifically this jumped out:
> provided that in both cases the URI is known in advance
which is not the case here

Reply:
👍